src/Microsoft.DotNet.ProjectModel/Compilation/LibraryExporter.cs

```diff
@@ -156,32 +155,57 @@ private LibraryExport GetExport(LibraryDescription library)
 
         private LibraryExport ExportPackage(PackageDescription package)
         {
-            var nativeLibraries = new List<LibraryAsset>();
-            PopulateAssets(package, package.Target.NativeLibraries, nativeLibraries);
-
-            var runtimeAssemblies = new List<LibraryAsset>();
-            PopulateAssets(package, package.Target.RuntimeAssemblies, runtimeAssemblies);
-
-            var compileAssemblies = new List<LibraryAsset>();
-            PopulateAssets(package, package.Target.CompileTimeAssemblies, compileAssemblies);
-
-            var sourceReferences = new List<string>();
-            foreach (var sharedSource in GetSharedSources(package))
+            var builder = LibraryExportBuilder.Create(package);
+            builder.WithNativeLibraries(PopulateAssets(package, package.Target.NativeLibraries));
+            builder.WithRuntimeAssemblies(PopulateAssets(package, package.Target.RuntimeAssemblies));
+            builder.WithCompilationAssemblies(PopulateAssets(package, package.Target.CompileTimeAssemblies));
+            builder.WithSourceReferences(GetSharedSources(package));
+            builder.WithAnalyzerReference(GetAnalyzerReferences(package));
+
+            if (package.Target.ContentFiles.Any())
             {
-                sourceReferences.Add(sharedSource);
+                var parameters = PPFileParameters.CreateForProject(_rootProject.Project);
+                Action<Stream, Stream> transform = (input, output) => PPFilePreprocessor.Preprocess(input, output, parameters);
+
+                var sourceCodeLanguage = _rootProject.Project.GetSourceCodeLanguage();
+                var languageGroups = package.Target.ContentFiles.GroupBy(file => file.CodeLanguage);
+                var selectedGroup = languageGroups.FirstOrDefault(g => g.Key == sourceCodeLanguage) ??
```

Comment:
We either take all content file for project language, or all for 'any` language.